scripts/extensions/sams/src/components/sets/setEditorPanel.tsx

```diff
@@ -0,0 +1,197 @@
+import * as React from 'react';
+import {cloneDeep, isEqual, pickBy} from 'lodash';
+
+import {ISuperdesk} from 'superdesk-api';
+import {ISamsAPI, ISet, ISetItem, IStorageDestinationItem} from '../../interfaces';
+import {SET_STATE} from '../../constants';
+
+import {Button, ButtonGroup, Input, Option, Select, Switch} from 'superdesk-ui-framework/react';
+
+import {
+    FormGroup,
+    FormRow,
+    PanelContent,
+    PanelContentBlock,
+    PanelContentBlockInner,
+    PanelHeader,
+    PanelHeaderSlidingToolbar,
+} from '../../ui';
+
+interface IProps {
+    set?: ISetItem;
+    destinations?: Array<IStorageDestinationItem>;
+    onDelete?(set: ISetItem): void;
+    onClose(): void;
+}
+
+interface IState {
+    diff: ISet;
+    isDirty?: boolean;
+    isExisting?: boolean;
+}
+
+export function getSetEditorPanel(superdesk: ISuperdesk, api: ISamsAPI) {
+    const {gettext} = superdesk.localization;
+
+    return class SetEditorPanel extends React.Component<IProps, IState> {
+        constructor(props: IProps) {
+            super(props);
+
+            if (this.props.set?._id == null) {
+                this.state = {
+                    diff: {
+                        destination_name: this.props.destinations?.[0]?._id,
+                    },
+                    isDirty: true,
+                    isExisting: false,
+                };
+            } else {
+                this.state = {
+                    diff: cloneDeep<ISetItem>(this.props.set),
+                    isDirty: false,
+                    isExisting: true,
+                };
+            }
+
+            this.onStateChange = this.onStateChange.bind(this);
+            this.onSave = this.onSave.bind(this);
+        }
+
+        onChange = {
+            name: (value: string) => this.onFieldChange('name', value),
+            description: (value: string) => this.onFieldChange('description', value),
+            destination_name: (value: string) => this.onFieldChange('destination_name', value),
+        };
+
+        onFieldChange(field: string, value: string): void {
+            const dirtyFields = ['state', 'name', 'description', 'destination_name'];
```

Comment:
diffs can be computed dynamically in a generic way without knowing the field names. Why hardcode them?

There's a library we are using for computing diffs https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/helpers/CrudManager.tsx#L105